docs/wiki/PipelineRemovalAction.md

```diff
@@ -0,0 +1,67 @@
+# Removal action
+
+This section describes how the removal of resources deployed by a module is performed and how to modify the default behaviour if a specific module or resource type needs it.
+
+---
+
+### _Navigation_
+
+- [Overview](#Overview)
+- [How it works](#how-it-works)
+- [Create a specialized removal procedure](#create-a-specialized-removal-procedure)
+
+---
+
+## Overview
+
+The Removal action is triggered after the deployment completes. This is used for several reasons:
+- Make sure to keep the validation subscription cost as low as possible.
+- Enable testing of the full module deployment at every run.
+
+The default removal procedure works fine for most of the modules created so far, so it's likely you won't have to change anything to make the module you're editing to be removed correctly after deployment.
+
+## How it works
+
+The removal process will remove all resources created during deployment. The list is identified by:
+
+1. Recursively fetching the list of resource IDs created through your deployment (resources created by deployments created by the parent one will be fetched too)
+1. Ordering the list based on resource IDs segment count (ensures child resources are removed first. E.g. `storageAccount/blobServices` comes before `storageAccount` as it has one more segments delimited by `/`)
+1. Filtering out from the list any resource used as dependencies for different modules (e.g. the commonly used Log Analytics workspace)
+1. Moving specific resource types to the top of the list (if a certain order is required). For example `vWAN` requires its `Virtual Hubs` to be removed first, even though they are no child-resources.
+
+After a resource is removed (this happens after each resource in the list), the script will execute, if defined, a **post removal operation**. This can be used for those resource types that requires a post processing, like purging a soft-deleted key vault.
+
+The procedure is initiated by the script `/utilities/pipelines/resourceRemoval/Initialize-DeploymentRemoval.ps1`, run during deployment by:
+- (Azure DevOps) `/.azuredevops/pipelineTemplates/module.jobs.deploy.yml`
+- (GitHub) `/.github/actions/templates/validateModuleDeployment/action.yml`
+
+It uses several helper scripts that can be found in the `/utilities/pipelines/resourceRemoval/helper` folder
+## Create a specialized removal procedure
+
+You can define a custom removal procedure by:
+1. influencing the **sequence** in which resource are removed by prioritizing specific resource types (_Virtual WAN_ and _Automation Account_ are examples of modules that uses this method)
```

Comment:
Good thing that you added the 3rd option, however, as per the original comment I'd suggest to move the example into a intended bullet for each path to allow us to explain 'why' e.g. an RSV would need a custom removal action. Just to say it needs one does not yet provide the full picture.

As a reference:
#1 is required for example for _Virtual WAN_ as it requires its connected resources to be removed first
#2 is required for example for _Recovery Services Vaults_ as they may contain protected items
#3 is required for example for _Key Vaults_ to purge them if they are soft-deleted

Reply:
Hmmm... I agree simply mentioning the resource type doesn't provide any additional value.
But I don't know... I would even remove the example, that are described above in the "how it works" section.
I prefer to have a clean list and don't repeat anything.
Maybe it's a personal thing, but for every additional second it takes to find what I need, chances I will abandon the reading dramatically increase

Reply:
But would the further indent not make clear that what ever comes after is an example one can skip/read over? I see your point and there is no clear answer.
Forthunatly, we have @eriqua to save the day ;)

Reply:
Not sure if I got the problem here but if the issue is readablity we may want to put the example in a note. For instance:

```suggestion
1. influencing the **order** in which resources are removed by prioritizing specific resource types 
    > **Example** Removing a _Virtual WAN_ resource requires related resources to be deleted in a specific order
```